src/main/java/org/jabref/JabRefGUI.java

```diff
@@ -188,6 +205,60 @@ private void saveWindowState(Stage mainStage) {
         Globals.prefs.putDouble(JabRefPreferences.POS_Y, mainStage.getY());
         Globals.prefs.putDouble(JabRefPreferences.SIZE_X, mainStage.getWidth());
         Globals.prefs.putDouble(JabRefPreferences.SIZE_Y, mainStage.getHeight());
+        printWindowState(mainStage);
+    }
+
+    /**
+     * outprints the Data from the Screen
+     * (only in debug mode)
+     * @param mainStage
+     */
+    private void printWindowState(Stage mainStage) {
+        StringBuilder bob = new StringBuilder();
+        bob.append("SCREEN DATA:");
+        bob.append("JabRefPreferences.WINDOW_MAXIMISED: " + mainStage.isMaximized() + "\n");
+        bob.append("JabRefPreferences.POS_X: " + mainStage.getX() + "\n");
+        bob.append("JabRefPreferences.POS_Y: " + mainStage.getY() + "\n");
+        bob.append("JabRefPreferences.SIZE_X: " + mainStage.getWidth() + "\n");
+        bob.append("JabRefPreferences.SIZE_Y: " + mainStage.getHeight() + "\n");
+        LOGGER.debug(bob.toString());
+    }
+
+    /**
+     * Tests if the window coordinates are out of the mainscreen
+     * @return outbounds
+     */
+    private boolean testExternalCoordinates() {
+        boolean outbounds = false;
+        if (Globals.prefs.getDouble(JabRefPreferences.POS_X) > Globals.prefs.getDouble(JabRefPreferences.SIZE_X)) {
+            outbounds = true;
+        }
+        if (Globals.prefs.getDouble(JabRefPreferences.POS_Y) > Globals.prefs.getDouble(JabRefPreferences.SIZE_Y)) {
+            outbounds = true;
+        }
+        if (Globals.prefs.getDouble(JabRefPreferences.POS_X) < 0) {
+            outbounds = true;
+        }
+        if (Globals.prefs.getDouble(JabRefPreferences.POS_Y) < 0) {
+            outbounds = true;
+        }
+        return outbounds;
+    }
+
+    /**
+     * returns the number of Monitors connected to the Computer
+     * @return numberOfmonitors
+     */
+    private int numberOfMonitors() {
```

Comment:
I think the pure JavaFX version would be `Screen.getScreens().size()`

Reply:
integrated with newest commit